docs/cadence_migration_guide/type-annotations-guide.mdx

```diff
@@ -0,0 +1,73 @@
+---
+title: Cadence Type Annotations 1.0 Migration Guide
+sidebar_position: 5
+sidebar_label: Cadence Type Annotations 1.0 Guide
+---
+
+# Type Annotations in Cadence 1.0
+
+In addition to updating your contracts in reaction to the Core Contract changes in Cadence 1.0, 
+certain language changes in Cadence 1.0 will also require changes to type annotations in your contracts, 
+in particular type annotations on resource and struct fields. 
+These type updates are required to accurately reflect the way that the Cadence 1.0 data migrations will change
+the types of these fields' values, and the Cadence 1.0 upgrade validator will enforce that these upgrades are accurate. 
+
+## Restricted Types
+
+In Cadence 1.0, support for restricted types was dropped, and replaced with intersection types. 
+As such, any existing restricted types must be replaced with a different type. 
+Restricted typed-values are migrated according to a specific set of rules, and type updates must also mirror this. 
+
+* `AnyStruct{I}` and `AnyResource{I}` should be migrated to just `{I}`, as these types have identical behavior
+* For any other type `T`, `T{I}` should be migrated to `T`, as this is the most specific possible type that can go here
+* For any type `T`, `T{}` should be migrated to just `T`, as there is no support for empty intersection types
+
+So, for example, a value of type `Vault{Receiver}` should be migrated to just a `Vault` value.
+
+## Reference Types
+
+Reference types from contracts written in Cadence v0.42 will need to be given entitlements in order to retain the same functionality 
```

Comment:
It would be good to clarify what you mean by "reference types" here

Reply:
In what sense? Are we expecting that Cadence developers won't be familiar with what a reference type is?

Reply:
I am pretty sure I know what a reference type is, but the way it is used in this sentence makes it sound like it is something else. When I hear reference types, I think of just a reference, like `&FlowToken.Vault`, but what I think you're saying here is that in addition to updating those, when there is a capability that has a specified type, like `Capability<&FlowToken.Vault{FungibleToken.Provider}>`, then the type of that capability needs to be given an entitlement to retain that functionality. 

I feel like just adding some of that context and an example would be really useful for developers who aren't programming language experts and don't have a clear understanding of programming language terminology like you do.

Reply:
Sure, that makes sense!